packages/dialtone-vue3/components/rich_text_editor/extensions/emoji/emoji.js

```diff
@@ -145,10 +121,31 @@ export const Emoji = Node.create({
   addProseMirrorPlugins () {
     return [
       Suggestion({
+        char: ':',
+        pluginKey: new PluginKey('emoji'),
         editor: this.editor,
         ...this.options.suggestion,
         ...suggestionOptions,
       }),
     ];
   },
+
+  addKeyboardShortcuts () {
+    return {
+      Backspace: () => this.editor.commands.command(({ tr, state }) => {
+        let isEmoji = false;
+        const { selection } = state;
+        const { empty, anchor } = selection;
+        if (!empty) { return false; }
+        state.doc.nodesBetween(anchor - 1, anchor, (node, pos) => {
+          if (node.type.name === this.name) {
+            isEmoji = true;
+            tr.insertText('', pos, pos + node.nodeSize);
+            return false;
+          }
+        });
+        return isEmoji;
+      }),
```

Comment:
I question whether this is a good way to handle this.. what if they press delete instead of backspace?

Reply:
It works, that function is to avoid "undo" while pressing backspace after entering an emoji.

![2024-09-05 12 35 40](https://github.com/user-attachments/assets/f4a618c4-d825-442f-8451-9651e5e42234)

Reply:
oh... strange.. That is not what I thought it was for haha